public/app/plugins/datasource/grafana/datasource.ts

```diff
@@ -135,14 +134,14 @@ export class GrafanaDatasource extends DataSourceWithBackend<GrafanaQuery> {
           buffer.maxDelta = request.range.to.valueOf() - request.range.from.valueOf();
         }
 
-        results.push(
-          getGrafanaLiveSrv().getDataStream({
-            key: `${request.requestId}.${counter++}`,
-            addr: addr!,
-            filter,
-            buffer,
-          })
-        );
+        const opts = {
+          key: `live-${target.refId}-`,
```

Comment:
should we also change it here? https://github.com/grafana/grafana/blob/963fffeef10cdad8276e46c6a20f2a13b0491c31/public/app/features/live/centrifuge/service.ts#L193

Reply:
right - but the change in the PR only applies to the Grafana DS, wouldn't we want to apply it to all ds?

Reply:
are there going to be collisions if two panels have streams with the same ref ID?

Reply:
not here -- each panel has its PanelQueryRunner that is managing the 'key'

@ArturWierzbicki -- i think you are right... we may want a better default for all datasources, but lets make sure it is fixed for the GrafanaDS before expanding scope